### PR TITLE
SYN-5870: optional user count limits to account for locked/archived users

### PR DESCRIPTION
**Pull request by MichaelSquires**
- Update code to account for locked/archived users
- Update test

### CODE REVIEW

File: synapse/lib/hiveauth.py [hunk 1]
```diff
@@ -413,6 +416,32 @@ def reqAuthGate(self, iden):
             raise s_exc.NoSuchAuthGate(iden=iden, mesg=mesg)
         return gate
 
+    def checkUserLimit(self):
+        '''
+        Check if we're at the specified user limit.
+
+        This should be called right before adding/unlocking/unarchiving a user.
+
+        Raises: s_exc_HitLimit if the number of active users is at the maximum.
```
Comment (Cisphyx) [inline synapse/lib/hiveauth.py:425]: got a "_" instead of "." in "s_exc_HitLimit"

Comment (MichaelSquires) [reply synapse/lib/hiveauth.py]: Fixed. Thanks!

**Review by Cisphyx — APPROVED** [review]

File: synapse/lib/cell.py [hunk 2]
```diff
@@ -2814,7 +2814,7 @@ def getCellNexsRoot(self):
 
     async def _initCellHiveAuth(self):
 
-        maxusers = self.conf.get('max:users')
+        maxusers = self.conf.get('max:users', 0)
```
Comment (vEpiphyte) [inline synapse/lib/cell.py:2817]: Is there a time when we would now have a `max:users` value when we have a `default` being set in confbase?

Comment (MichaelSquires) [reply synapse/lib/cell.py:2817]: Yep, this test: https://github.com/vertexproject/synapse/blob/blackout/SYN-5870/user-count-checks-archived-locked/synapse/tests/test_lib_config.py#L248

Comment (vEpiphyte) [reply synapse/lib/cell.py:2817]: That makes me think there is a separate error in the config stuff if the default isn't being set. Checking.

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**